epoxy-processor/src/main/java/com/airbnb/epoxy/ControllerModelField.kt

```diff
@@ -0,0 +1,36 @@
+package com.airbnb.epoxy
+
+import com.squareup.javapoet.TypeName
+
+import javax.lang.model.element.Element
+
+import com.airbnb.epoxy.Utils.isFieldPackagePrivate
+
+class ControllerModelField(element: Element) {
+
+    var fieldName: String = element.simpleName.toString()
+    var typeName: TypeName = TypeName.get(element.asType())
+    var packagePrivate: Boolean = isFieldPackagePrivate(element)
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
The thing is that data class primary constructor must have only property parameters so if we want this class as a data class we need to change its constructor and init properties on the caller side. If you think that it makes sense - we can change it.